embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionHandler.kt

```diff
@@ -241,32 +252,12 @@
     /**
      * It determines if we are allowed to build an end session message.
      */
-    private fun isAllowedToEnd(endType: LifeEventType, activeSession: Session): Boolean {
-        return when (endType) {
-            LifeEventType.STATE -> {
-                // state sessions are always allowed to be ended
-                logger.logDebug("Session is STATE, it is always allowed to end.")
-                true
-            }
-
-            LifeEventType.MANUAL -> {
-                if (!configService.sessionBehavior.isSessionControlEnabled()) {
-                    logger.logWarning(
-                        "Session control disabled from remote configuration. " +
-                            "Session is not allowed to end."
-                    )
-                    false
-                } else if ((clock.now() - activeSession.startTime) < minSessionTime) {
-                    // If less than 5 seconds, then the session cannot be finished manually.
-                    logger.logError("The session has to be of at least 5 seconds to be ended manually.")
-                    false
-                } else {
-                    logger.logDebug("Session allowed to end.")
-                    true
-                }
-            } else -> false // background activity
```

Comment:
The `SessionHandler` handling a background activity case should've told us there's some leakiness of this abstraction....